### PR TITLE
test automation for 'schedule' feature

### PR DESCRIPTION
**Pull request by leenaj0hn**

### CODE REVIEW

File: tests/integration/hpe_3par_manager.py [hunk 1]
```diff
@@ -439,7 +452,7 @@ def hpe_verify_volume_created(self, volume_name, vvs_name=None, **kwargs):
         if 'size' in kwargs:
             self.assertEqual(hpe3par_volume['sizeMiB'], int(kwargs['size']) * 1024)
         else:
-            self.assertEqual(hpe3par_volume['sizeMiB'], 102400)
+            self.assertEqual(hpe3par_volume['sizeMiB'], 10240)
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_manager.py]: Please confirm if this change is required

File: tests/integration/hpe_3par_manager.py [hunk 2]
```diff
@@ -746,6 +759,22 @@ def hpe_remove_vvs_qos(self, vvs_name):
...
+        hpe3par_schedule = hpe3par_cli.getSchedule(schedule_name)
+        #schedule_details = list(snapshot['Options'].values())
+        schedule_details_cli = hpe3par_schedule[2].split(',' , 11)
+        
+        self.assertEqual(snapshot['Options']['scheduleName'], schedule_details_cli[0])
+        self.assertEqual(snapshot['Options']['scheduleFrequency'], ' '.join(map(str,schedule_details_cli[2:7])))
+        schedule_command = schedule_details_cli[1].split(' ', 8) 
+        self.assertEqual(schedule_command[3], (snapshot['Options']['expHrs'])+"h")
+        self.assertEqual(schedule_command[5], (snapshot['Options']['retHrs'])+"h")
+        #self.assertEqual(schedule_command[3], exp_hrs)
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_manager.py]: Remove commented lines if unnecessary

File: tests/integration/hpe_3par_manager.py (hunk 1 above)
Comment (leenaj0hn) [reply tests/integration/hpe_3par_manager.py]: Reverted back

File: tests/integration/hpe_3par_manager.py [hunk 3]
```diff
@@ -746,6 +759,22 @@ def hpe_remove_vvs_qos(self, vvs_name):
         hpe3par_cli.deleteVolumeSet(vvs_name)
         hpe3par_cli.logout()
 
+    def hpe_verify_snapshot_schedule(self, schedule_name, snapshot):
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        hpe3par_cli = self._hpe_get_3par_client_login()
 
+        hpe3par_schedule = hpe3par_cli.getSchedule(schedule_name)
+        #schedule_details = list(snapshot['Options'].values())
+        schedule_details_cli = hpe3par_schedule[2].split(',' , 11)
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_manager.py]: Make this generic

File: tests/integration/hpe_3par_manager.py (hunk 2 above)
Comment (leenaj0hn) [reply tests/integration/hpe_3par_manager.py]: Removed commented lines

File: tests/integration/hpe_3par_manager.py (hunk 3 above)
Comment (leenaj0hn) [reply tests/integration/hpe_3par_manager.py]: Made splitting of the string generic

File: tests/integration/hpe_3par_api_schedule_test.py [hunk 4]
```diff
@@ -144,6 +144,9 @@ def test_create_schedule(self):
         self.hpe_inspect_snapshot(snapshot, snapshot_name=snapshot_name,
                                   virtualCopyOf=volume['Name'], size=THIN_SIZE)
 
+       	import pdb
+        pdb.set_trace()
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_api_schedule_test.py]: Remove pdb

File: tests/integration/hpe_3par_manager.py [hunk 5]
```diff
@@ -145,6 +145,9 @@ def hpe_inspect_volume(self, volume, **kwargs):
                 self.assertEqual(inspect_volume['Status']['qos_detail'][option], kwargs[option])
             else:
                 pass
+        if (self.assertIn('Snapshots', inspect_volume['Status'])):
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_manager.py]: Cannot use this assertIn in if loop

File: tests/integration/hpe_3par_manager.py [hunk 6]
```diff
@@ -222,6 +225,15 @@ def hpe_inspect_snapshot(self, snapshot, **kwargs):
         else:
             self.assertEqual(inspect_snapshot['Status']['snap_detail']['snap_cpg'],
                              SNAP_CPG)
+        if (self.assertIn('snap_schedule', inspect_snapshot['Status']['snap_detail'])):
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_manager.py]: Cannot use this assertIn in if loop

File: tests/integration/hpe_3par_api_schedule_test.py (hunk 4 above)
Comment (leenaj0hn) [reply tests/integration/hpe_3par_api_schedule_test.py]: removed pdb

File: tests/integration/hpe_3par_manager.py (hunk 5 above)
Comment (leenaj0hn) [reply tests/integration/hpe_3par_manager.py]: removed assertIn in 'if'

File: tests/integration/hpe_3par_manager.py (hunk 6 above)
Comment (leenaj0hn) [reply tests/integration/hpe_3par_manager.py]: removed assertIn in 'if'

File: tests/integration/hpe_3par_manager.py [hunk 7]
```diff
@@ -111,6 +111,7 @@ def hpe_inspect_volume(self, volume, **kwargs):
 
         volume_details = ['size', 'provisioning', 'flash_cache', 'compression', 'mountConflictDelay', 'importVol', 'cpg', 'snapcpg']
         qos_detail = ['enabled', 'maxIOPS', 'minIOPS', 'priority', 'vvset_name']
+        snapshots = inspect_volume['Status']['Snapshots'][0]
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_manager.py]: This will throw keyerror with other tests

Comment (leenaj0hn) [reply tests/integration/hpe_3par_manager.py]: added checks specific to spashots

File: tests/integration/hpe_3par_api_schedule_test.py [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+                    })
+                else:
+                    c.configure_plugin(HPE3PAR, {
+                        'certs.source': CERTS_SOURCE,
+                        'glibc_libs.source': '/lib64'
+                    })
+                pl_data = c.inspect_plugin(HPE3PAR)
+                assert pl_data['Enabled'] is False
+                while pl_data['Enabled'] is False:
+                    c.enable_plugin(HPE3PAR)
```
Comment (shaik-gousa-khaja-asif) [inline tests/integration/hpe_3par_api_schedule_test.py]: Need to wait for backend to initialize . Use def hpe_wait_for_all_backends_to_initialize()

**Pull request merged**